### PR TITLE
Connect: Refresh leaf cluster certs before fetching certs for database

### PR DESCRIPTION
**Pull request by ravicious**
Fixes gravitational/webapps.e#215.

tl;dr It was impossible to use Connect with databases on leaf clusters. Any attempt to create a db proxy were failing with "SSH cert not available" when Connect was trying to store reissued db certs.

Roman pointed out that the SSH key is indeed missing when looking at the tsh directory tree created by Connect. Connect was not doing the equivalent of what `tsh login <cluster>` was doing.

The fix is to reissue certs for the cluster first before attempting to reissue certs for the db.

---

Another issue I encountered is that when working with a leaf cluster, we used to create a TeleportClient instance aimed at the root cluster and only after it was created we updated `TeleportClient.SiteName` to point at the leaf cluster. This didn't update the site name in `TeleportClient.LocalKeyAgent`.

While it wasn't causing any problems yet, I decided to fix it while I was in this part of the codebase. So the first commit in this PR actually makes sure to pass the correct `SiteName` before we create an instance of TeleportClient.

### CODE REVIEW

File: lib/teleterm/clusters/storage.go [hunk 1]
```diff
@@ -161,19 +155,27 @@ func (s *Storage) addCluster(ctx context.Context, dir, webProxyAddress string) (
 }
 
 // fromProfile creates a new cluster from its profile
-func (s *Storage) fromProfile(clusterName string) (*Cluster, error) {
-	if clusterName == "" {
+func (s *Storage) fromProfile(profileName, leafClusterName string) (*Cluster, error) {
```
Comment (ravicious) [inline lib/teleterm/clusters/storage.go:158]: There's an outstanding issue in Teleport Connect where all our URIs use the proxy hostname as the cluster name (which is equivalent to a profile name, I think). I'll create a separate ticket for it soon, but in general the problem is that the proxy hostname can be different than the root cluster name.

So when `Storage.GetByURI` calls `fromProfile`, this name is actually the profile name and not the real root cluster name from the key. I changed the argument name here from `clusterName` to `profileName` to signal that for the future.

File: lib/teleterm/clusters/storage.go [hunk 2]
```diff
@@ -202,10 +204,15 @@ func (s *Storage) fromProfile(clusterName string) (*Cluster, error) {
 		return nil, trace.Wrap(err)
 	}
 
-	clusterURI := uri.NewClusterURI(clusterName)
+	clusterURI := uri.NewClusterURI(profileName)
+	if leafClusterName != "" {
+		clusterURI = clusterURI.AppendLeafCluster(leafClusterName)
+	}
```
Comment (ravicious) [inline lib/teleterm/clusters/storage.go:210]: Previously this was done in `Storage.GetByURI`, but I see no reason why we couldn't do it here.

File: lib/teleterm/clusters/storage.go [hunk 3]
```diff
@@ -202,10 +204,15 @@ func (s *Storage) fromProfile(clusterName string) (*Cluster, error) {
...
 	return &Cluster{
-		URI:           clusterURI,
-		Name:          clusterName,
+		URI: clusterURI,
+		// TODO(ravicious): This should probably use leafClusterName if available, but at this point I'm
+		// worried that changing it might break something else in the app.
+		Name:          profileName,
```
Comment (ravicious) [inline lib/teleterm/clusters/storage.go:215]: At the moment it doesn't cause any issues because Teleport Connect uses [another code path](https://github.com/gravitational/teleport/blob/d9deaff3d4465bd7654a0d698475d90b2b2d87fe/lib/teleterm/clusters/cluster_leaves.go#L41) to get names of leaf clusters.

Methods on `Storage` are used mostly to get a `Cluster` instance that lets us perform operations on the root/leaf cluster through `TeleportClient`.

File: lib/teleterm/clusters/storage.go [hunk 4]
```diff
@@ -161,19 +155,27 @@ func (s *Storage) addCluster(ctx context.Context, dir, webProxyAddress string) (
...
 // fromProfile creates a new cluster from its profile
-func (s *Storage) fromProfile(clusterName string) (*Cluster, error) {
-	if clusterName == "" {
+func (s *Storage) fromProfile(profileName, leafClusterName string) (*Cluster, error) {
+	if profileName == "" {
 		return nil, trace.BadParameter("cluster name is missing")
 	}
 
+	clusterNameForKey := profileName
+	if leafClusterName != "" {
```
Comment (zmb3) [inline lib/teleterm/clusters/storage.go:164]: There's 3 separate checks for `if leafClusterName != ""` - can we combine them?

**Review by zmb3 — APPROVED** [review]

**Review by nklaassen — APPROVED** [review]
